sbc_compassion/models/correspondence_template.py

```diff
@@ -168,7 +175,76 @@ def _set_image(self):
                 })
                 self.template_attachment_id = attachment.id
 
-    @api.depends('template_image')
+    @api.onchange('template_image')
+    def _onchange_template_image(self):
+        for template in self:
+            if template.template_image:
+                with tempfile.NamedTemporaryFile() as template_file:
+                    data = template.with_context(bin_size=False).template_image
+                    if not data:
+                        data = template.template_image
+                    template_file.write(base64.b64decode(
+                        data))
+                    template_file.flush()
+                    # Find the pattern inside the template image
```

Comment:
Here you say you find the pattern, but method `_compute_img_constant` seems to compute only qr code and page size.